### PR TITLE
Add ability to mount secrets in the PodSpec builder

### PR DESCRIPTION
**Pull request by sshardool**

### CODE REVIEW

File: azkaban-common/src/main/java/azkaban/container/models/AzKubernetesV1SpecBuilder.java [hunk 1]
```diff
@@ -180,6 +182,33 @@ public AzKubernetesV1SpecBuilder addHostPathVolume(final String volName, final S
...
+     * @param volName volume name
+     * @param secretName secret name
+     * @param volMountPath directory where secret will be mounted
+     */
+    public AzKubernetesV1SpecBuilder addSecretVolume(final String volName,
+        final String secretName, final String volMountPath) {
+        final V1SecretVolumeSource secretVolumeSource =
+            new V1SecretVolumeSourceBuilder()
+                .withNewSecretName(secretName)
+                .withDefaultMode(0400) //set read permissions only for the the pod user
```
Comment (djaiswal83) [inline azkaban-common/src/main/java/azkaban/container/models/AzKubernetesV1SpecBuilder.java:197]: Can we make "0400" a constant?

File: azkaban-common/src/test/java/azkaban/container/models/AzKubernetesV1PodBuilderTest.java [hunk 2]
```diff
@@ -49,6 +49,8 @@ public void testPodSpec() throws IOException {
                 azConfVer)
             .addHostPathVolume("nscd-socket", "/var/run/nscd/socket", "Socket",
                 "/var/run/nscd/socket")
+            .addSecretVolume("azkaban-private-properties", "holdem6-azkaban-private-properties",
```
Comment (djaiswal83) [inline azkaban-common/src/test/java/azkaban/container/models/AzKubernetesV1PodBuilderTest.java:52]: Can you remove reference to Li-internal names, applies to yaml file as well?

**Review by djaiswal83 — CHANGES_REQUESTED** [review]
The code looks good. I have few comments.

File: azkaban-common/src/test/java/azkaban/container/models/AzKubernetesV1PodBuilderTest.java (hunk 2 above)
Comment (sshardool) [reply azkaban-common/src/test/java/azkaban/container/models/AzKubernetesV1PodBuilderTest.java]: Good point. Fixed.

File: azkaban-common/src/main/java/azkaban/container/models/AzKubernetesV1SpecBuilder.java (hunk 1 above)
Comment (sshardool) [reply azkaban-common/src/main/java/azkaban/container/models/AzKubernetesV1SpecBuilder.java:197]: These are standard file permissions (therefore meaning of 0400 is well understood) and this is the only location this value is used. 
Defining a separate constant for it seemed like an overkill. Please let me know if this is still a concern.

Comment (jakhani) [inline azkaban-common/src/main/java/azkaban/container/models/AzKubernetesV1SpecBuilder.java:197]: Can you please create constant for permission?

File: azkaban-common/src/test/resources/azkaban/container/models/v1PodTest1.yaml [hunk 3]
```diff
@@ -51,3 +53,7 @@ spec:
       type: Socket
       path: /var/run/nscd/socket
     name: nscd-socket
+  - name: azkaban-private-properties
+    secret:
+      defaultMode: 256
+      secretName: holdem6-azkaban-private-properties
```
Comment (jakhani) [inline azkaban-common/src/test/resources/azkaban/container/models/v1PodTest1.yaml]: Nit: New line in the end.

File: azkaban-common/src/main/java/azkaban/container/models/AzKubernetesV1SpecBuilder.java (hunk 1 above)
Comment (sshardool) [reply azkaban-common/src/main/java/azkaban/container/models/AzKubernetesV1SpecBuilder.java]: Fixed. (was pointed out in another comment as well so decided to change this)

Comment (sshardool) [reply azkaban-common/src/main/java/azkaban/container/models/AzKubernetesV1SpecBuilder.java]: Fixed.

File: azkaban-common/src/test/resources/azkaban/container/models/v1PodTest1.yaml (hunk 3 above)
Comment (sshardool) [reply azkaban-common/src/test/resources/azkaban/container/models/v1PodTest1.yaml]: Fixed.

**Review by djaiswal83 — DISMISSED** [review]
Thanks for addressing review comments. LGTM

**Review by jakhani — APPROVED** [review]

**Review by djaiswal83 — APPROVED** [review]

**Pull request merged**